### PR TITLE
Sort by a new `sticky` parameter

### PR DESCRIPTION
**Pull request by stevenjoezhang**
Test cases / docs will be updated later.

Node.js 12+ uses stable sorting algorithms by default, thus `lodash.orderBy` is only used for Node.js 10 compatibility.
https://v8.dev/features/stable-sort

```js
// Node.js 10
posts.data = _.orderBy(posts.data, post => post.sticky || 0, "desc");

// Node.js 12+
posts.data.sort((a, b) => (b.sticky || 0) - (a.sticky || 0));
```

### CODE REVIEW

**Comment by SukkaW** [comment]
https://github.com/mziccard/node-timsort

There is high performance tim-sort implementation that could be used as an alternative.

**Comment by stevenjoezhang** [comment]
Another approach is

```js
const posts = locals.posts.sort(`${config.index_generator.order_by} -sticky`);
```

However, the current implementation of warehouse does not support setting default values for sort (e.g. sticky = 0), thus the sorting will not take effect unless every post has a `sticky` or `order` attribute.

https://github.com/hexojs/hexo-generator-index/blob/88843c2ae12d7ca96865de720e1dffbe96bb5252/test/index.js#L20-L22

https://github.com/hexojs/warehouse/blob/c24cf938152ea0c3fd04a9bac0ca1db8e661bdb2/lib/schema.js#L60

File: README.md [hunk 1]
```diff
@@ -36,6 +37,18 @@ index_generator:
...
 
+## Usage
+
+The `sticky` parameter in the post [Front-matter](https://hexo.io/docs/front-matter) will be used to pin the post to the top of the index page.
+
+```markdown
+---
+title: Hello World
+date: 2013/7/13 20:46:25
+sticky: 100
```
Comment (curbengh) [inline README.md:48]: @stevenjoezhang 
lower value = higher priority?

Comment (stevenjoezhang) [reply README.md:48]: higher value = higher priority
```
% node
Welcome to Node.js v14.7.0.
Type ".help" for more information.
> const a = [1,2,3,4,5]
[ 1, 2, 3, 4, 5 ]
> a.sort((a,b)=>b-a)
[ 5, 4, 3, 2, 1 ]
>
```

**Pull request merged**